### PR TITLE
feat: date filter

### LINKED ISSUE #3: Feat: add before / after filtering
- [x] add "before" / "after" parameters

------

After that, the usage of this project can be:

```ts
import getLodestoneNews from 'lodestone-news'

const news = await getLodestoneNews({
  region: 'jp',
  category: 'topics',
  count: 10,
  before: new Date(1656633600 * 1000) // 2022-07-01 00:00:00 GMT+0
})

news // an array contains 10 news before 2022-07-01 00:00:00 GMT+0
```

### PR DESCRIPTION
**Pull request by SaarChaffee**
close #3

### CODE REVIEW

File: __tests__/index.spec.ts
Comment (MaikoTan) [inline __tests__/index.spec.ts]:
```suggestion
it('should filter by date', async () => {
```

File: __tests__/index.spec.ts
Comment (MaikoTan) [inline __tests__/index.spec.ts]: Add a comment explains that what time is this.
And it is better to test within a shorter duration.

File: src/news.ts
Comment (MaikoTan) [inline src/news.ts]:
```suggestion

```

no need to be class fields

File: src/news.ts
Comment (MaikoTan) [inline src/news.ts]:
```suggestion

```

File: src/utils.ts
Comment (MaikoTan) [inline src/utils.ts:20]: You can set a default value, for example, `before = new Date()` `after = 0`

**Review by MaikoTan — CHANGES_REQUESTED** [review]
Not 'Add' but 'feat' (means feature).

File: src/news.ts
Comment (SaarChaffee) [inline src/news.ts:60]: #

File: src/news.ts
Comment (SaarChaffee) [reply src/news.ts]: `before`、`after`作为一个可选项，应与`request`保持一致

File: src/news.ts
Comment (SaarChaffee) [reply src/news.ts]: 理由同上

File: src/utils.ts
Comment (SaarChaffee) [reply src/utils.ts:20]: `filter`不一定会被调用所以不需要默认值

**Review by SaarChaffee — COMMENTED** [review]
更改了样例命名与测试时间的区间

File: src/news.ts
Comment (MaikoTan) [reply src/news.ts]: Well, then the class-invoking style should be:

```ts
const lodestone = new LodestoneNews({ before: xxx, after: yyy })
// ...
lodestone.getNews() // get news between xxx and yyy *implicitly*
lodestone.getNews({ after: xxx }) // get nothing QAQ
```

It is weird to call a function without arguments but get data that filtered by some invisible mechanics.
So I insist to add `before` and `after` on `getNews` function, instead of being class fields.

File: src/news.ts
Comment (SaarChaffee) [inline src/news.ts:47]: 这里删掉58行会报错，因此改为从`option`解构出来

**Comment by MaikoTan** [comment]
Report an error if `becore` was less than `after`, then this can be merged after approval.

File: src/utils.ts
Comment (MaikoTan) [inline src/utils.ts]:
```suggestion
        throw new Error(`\`before\`(${before}) should not be less than \`after\`(${after}).`)
```

**Review by MaikoTan — CHANGES_REQUESTED** [review]

File: __tests__/index.spec.ts
Comment (MaikoTan) [inline __tests__/index.spec.ts]: - It is better to set `1658217599 * 1000` as a variable, and then use it here and below.

File: __tests__/index.spec.ts
Comment (MaikoTan) [inline __tests__/index.spec.ts]:
```suggestion
it('should filter with after / before', async () => {
```

File: __tests__/index.spec.ts
Comment (MaikoTan) [inline __tests__/index.spec.ts]: - set `1658217599 * 1000` as a variable that mentioned above
- `news[news.length - 1].date` can be simplified as `news.pop().date` or something, also that below

File: __tests__/index.spec.ts
Comment (MaikoTan) [inline __tests__/index.spec.ts]:
```suggestion
it('should filter with after', async () => {
```

File: __tests__/index.spec.ts
Comment (MaikoTan) [inline __tests__/index.spec.ts]: - Why not just call `getNews` as above.
```suggestion
  const news = getNews({
    category: 'topics',
    region: 'na',
    count: 100,
    after: new Date(1658217599 * 1000), // 2022-07-19 07:59:59 UTC+0
  })
```

File: src/utils.ts
Comment (MaikoTan) [inline src/utils.ts:5]:
```suggestion

import { News } from './config'
```

- should append one line.

File: __tests__/index.spec.ts
Comment (MaikoTan) [inline __tests__/index.spec.ts]: - This test suite is redundant, as it should be covered above.

File: __tests__/index.spec.ts
Comment (MaikoTan) [inline __tests__/index.spec.ts:70]: - These lines has been duplicated in every test, maybe you can extract them in a new function.

**Review by MaikoTan — CHANGES_REQUESTED** [review]
last several changes request

**Comment by SaarChaffee** [comment]
> ### Add
> * date filter ([e67cf12](https://github.com/AwesomeHamster/lodestone-news/commit/e67cf124aa1fd1ee57f6000ec7bf6b07b24f0b9b))
> * date filter test ([35c3fa0](https://github.com/AwesomeHamster/lodestone-news/commit/35c3fa00be80fab6b0aa5f77881ca70301f0e980))
> 
> ### Upd
> * case name ([ff2a8c6](https://github.com/AwesomeHamster/lodestone-news/commit/ff2a8c6d74f0498035a00ed0f6ec63407239d4ac))
> * test case ([a5a7fa8](https://github.com/AwesomeHamster/lodestone-news/commit/a5a7fa8d4db98e48c40454b7d0543844f593e488))
> * error msg format ([316fd82](https://github.com/AwesomeHamster/lodestone-news/commit/316fd82063b8f803ba5e3acec1febef9dc3cf2cf))
> 
> ### Fear
> * throwing filter errors ([be3e79a](https://github.com/AwesomeHamster/lodestone-news/commit/be3e79a528b18730bf44b57eb15458d564236cb3))
> 
> ### Features
> * judgment functions of news ([c098729](https://github.com/AwesomeHamster/lodestone-news/commit/c098729c11c1418a82715deee7ba652163b3f73c))
> * judgment functions of news ([dc14dec](https://github.com/AwesomeHamster/lodestone-news/commit/dc14dec3524d4e8170775827b42959ea21c4f1c6))
> 
> ### Opt
> * way of judgement news list ([0a1e994](https://github.com/AwesomeHamster/lodestone-news/commit/0a1e994cb10714c8685d67e751b4c60c91334a15))
> 
> ### Code Refactoring
> * migrating functions to unit test ([64db2c7](https://github.com/AwesomeHamster/lodestone-news/commit/64db2c7b6a83fa45453e01be855b0bad7948d849))
> 
> ### Contributors
> [SaarChaffee](https://github.com/SaarChaffee)
> 
> Commit-Lint commands
> 
> You can trigger Commit-Lint actions by commenting on this PR:
> 
> * `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
> * `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
> * `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
> * `@Commit-Lint merge disable` will desactivate merge dependabot PR
> * `@Commit-Lint review` will approve dependabot PR
> * `@Commit-Lint stop review` will stop approve dependabot PR

@Commit-Lint review

**Comment by SaarChaffee** [comment]
@Commit-Lint stop review

**Comment by SaarChaffee** [comment]
@Commit-Lint stop review

**Review by MaikoTan — APPROVED** [review]

**Pull request merged**
